src/test/java/pages/ProductsPage.java

```diff
@@ -0,0 +1,102 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductsPage extends BasePageLoggedIn {
+
+    String itemContainer = "//div[@class='inventory_item_name' and text()='%s']/ancestor::div[@class='inventory_item']";
+    By selectLocator = By.cssSelector("select[class='product_sort_container']");
+    By headerContainer = By.id("header_container");
+    By itemDetailsLink = By.cssSelector("a[id$='title_link']");
+    By addRemoveButton = By.tagName("button");
+    By addToCartButton = By.cssSelector("button[id^='add-to-cart]");
+    By removeButton = By.cssSelector("button[@id^='remove-']");
+    By productPrice = By.cssSelector("div.inventory_item_price");
+    By productDescription = By.cssSelector("div.inventory_item_desc");
+
+    public ProductsPage(WebDriver driver) {
+        super(driver);
+    }
+
+
+    public boolean isHeaderContainerDisplayed() {
+        return driver.findElement(headerContainer).isDisplayed();
+    }
+
+    public void clickAddRemoveButton(int productId) {
+        getAllProductContainers().get(productId).findElement(addRemoveButton).click();
+    }
+
+    public void addRemoveAllProducts() {
+        for (int i = 0; i < 6; i++) {
```

Comment:
What if products count changes to any other number?
It's better to iterate through all actually avaliable products at the moment e.g.
for (product of getAllProductContainers()) {
product.findElement(addRemoveButton).click();
}

Reply:
Ok, will update methods.